index.html

```diff
@@ -722,47 +1502,116 @@ <h2>
+            <li>If |gamepad|'s [=relevant global object=] is a {{Window}}
+            object, then [=queue a global task=] on the [=gamepad task source=]
+            to [=fire an event=] named {{gamepadconnected}} at |gamepad|'s
+            [=relevant global object=] using {{GamepadEvent}} with its
+            {{GamepadEvent/gamepad}} attribute initialized to |gamepad|.
+            </li>
+          </ol>
+        </li>
+        <li>Set
+        |navigator|.{{Navigator/[[gamepads]]}}[|gamepad|.{{Gamepad/index}}] to
+        |gamepad|.
+        </li>
+      </ol>
       <p>
         User agents implementing this specification must provide a new DOM
-        event, named <code>gamepadconnected</code>. The corresponding event
-        MUST be of type {{GamepadEvent}} and, if <a>allowed to use</a> the
-        "`gamepad`" permission, MUST fire on the <code>window</code> object.
-        Registration for and firing of the <code>gamepadconnected</code> event
-        MUST follow the usual behavior of DOM Events. [[DOM]]
+        event, named {{gamepadconnected}}. The corresponding event MUST be of
+        type {{GamepadEvent}} and, if [=allowed to use=] the `"gamepad"`
+        permission, MUST fire on the {{Window}} object. Registration for and
+        firing of the {{gamepadconnected}} event MUST follow the usual behavior
+        of DOM Events. [[DOM]]
       </p>
       <p>
-        A <a>user agent</a> MUST dispatch this event type to indicate the user
-        has connected a gamepad. If a gamepad was already connected when the
-        page was loaded, the <a>gamepadconnected</a> event SHOULD be dispatched
-        when the user presses a button or moves an axis.
+        A [=user agent=] MUST dispatch this event type to indicate the user has
+        connected a gamepad. If a gamepad was already connected when the page
+        was loaded, the {{gamepadconnected}} event SHOULD be dispatched when
+        the user presses a button or moves an axis.
       </p>
       <p>
-        A <a>user agent</a> MUST NOT dispatch this event type if the
-        <a>environment settings object</a> is a <a>non-secure context</a>.
+        A [=user agent=] MUST NOT dispatch this event type if the [=environment
+        settings object=] is a [=non-secure context=].
       </p>
     </section>
     <section>
       <h3 id="event-gamepaddisconnected">
         The <dfn class="event">gamepaddisconnected</dfn> event
       </h3>
+      <p>
+        When a gamepad becomes unavailable on the system, run the following
+        steps:
+      </p>
+      <ol>
+        <li>If the [=current settings object=]'s [=environment settings object
+        / responsible document=] is not [=allowed to use=] the `"gamepad"`
+        permission, then abort these steps.
+        </li>
+        <li>Let |gamepad:Gamepad| be the {{Gamepad}} representing the
+        unavailable device.
+        </li>
+        <li>Set |gamepad|.{{Gamepad/[[connected]]}} to `false`.
```

Comment:
I think this should be done in the steps of the task (right before the event is fired), no? Otherwise, it would update somewhat randomly.

Reply:
Agreed. We should update the gamepad state synchronously with firing the event to make sure the state changes are handled sequentially and predictably.

I think this order makes the most sense for gamepadconnected:

1. Create gamepad with gamepad.connected initialized to true
2. Add gamepad to navigator.[[gamepads]]
3. Dispatch gamepadconnected

And for gamepaddisconnected:

1. Set gamepad.connected to false
2. Dispatch gamepaddisconnected
3. Remove gamepad from navigator.[[gamepads]]

Updating in this order ensures the gamepad.connected state matches the stated reflected by the gamepadconnected or gamepaddisconnected event, and also ensures the item in the getGamepads() array at index gamepad.index is the same instance as the gamepad instance in the event.